staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/defaulting/algorithm.go

```diff
@@ -34,20 +39,26 @@ func Default(x interface{}, s *structuralschema.Structural) {
 			if prop.Default.Object == nil {
 				continue
 			}
-			if _, found := x[k]; !found {
+			if _, found := x[k]; !found || isNonNullableNull(x[k], &prop) {
 				x[k] = runtime.DeepCopyJSONValue(prop.Default.Object)
 			}
 		}
-		for k, v := range x {
+		for k := range x {
 			if prop, found := s.Properties[k]; found {
-				Default(v, &prop)
+				Default(x[k], &prop)
 			} else if s.AdditionalProperties != nil {
-				Default(v, s.AdditionalProperties.Structural)
+				if isNonNullableNull(x[k], s.AdditionalProperties.Structural) {
+					x[k] = runtime.DeepCopyJSONValue(s.AdditionalProperties.Structural.Default.Object)
+				}
```

Comment:
this case is interesting. Where would we drop these nulls if there is no default? Would we drop them? It's missing in the KEP!

Reply:
I don't think that's a different case. Whether we use a named property or an additional property shouldn't make a difference?

Reply:
I agree that it is the same, but deserves a note in the KEP.